.github/workflows/release.yml

```diff
@@ -0,0 +1,64 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  check-token:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check token variable
+        env:
+          PYPI_TOKEN_VAR: ${{ secrets.pypi_token }}
+        run: |
+          if [ -z "$PYPI_TOKEN_VAR" ]; then
+            echo "Pypi token variable is empty!"
+            exit 1
+          fi
+          echo "Pypi token variable is set!"
+  build:
```

Comment:
Given that we have to run only a single build lets combine this job and `pypi-publish` to `deploy`. That way we also don't have to cache the generated artifacts and download them again for the next job.

Reply:
I think removing the artifact quickly(5min) might be a solution. Your suggestion is more appropriate for this project.